### PR TITLE
OHRM5X-1373: Develop shortlist & view action history screen

### PR DESCRIPTION
**Pull request by Ajiharan007**

### CODE REVIEW

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+<template>
+  <div class="orangehrm-background-container orangehrm-save-candidate-page">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        Shortlist Candidate
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:24]: please add translations here and all places.

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                :rules="rules.note"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button display-type="ghost" :label="$t('general.back')" />
+          <submit-button label="Shortlist" />
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:80]: Need space between buttons. please check other places and use same classes

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                type="textarea"
+                :rules="rules.note"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button display-type="ghost" :label="$t('general.back')" />
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:79]: Please add navigate action

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        notes: [shouldNotLessThanCharLength(250)],
+      },
+      shortlist: {
+        candidate: '',
+        vacancy: '',
+        manager: '',
+        status: '',
+        note: '',
+      },
+      statuses: [
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: is this not used in select?

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        },
+      ],
+    };
+  },
+  beforeMount() {
+    this.isLoading = true;
+    this.http.getAll().then(({data: {data}}) => {
+      const {candidate, status, ...rest} = data;
+      const {firstName, lastName, middleName} = candidate;
+      const fullName = `${firstName} ${middleName} ${lastName}`;
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:152]: Candidate can be deleted or purged, therefor name might not be always available please consult @devishke-orange

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+          navigate('/recruitment/viewCandidates');
+        });
+    },
+  },
+};
+</script>
+<style scoped lang="scss">
+.orangehrm-save-candidate-page {
+  &-full-width {
+    grid-column: 1 / span 2;
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:179]: why does this needed?

File: src/plugins/orangehrmRecruitmentPlugin/Controller/ShortlistCandidateController.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * @inheritDoc
+     */
+    public function preRender(Request $request): void
+    {
+        $component = new Component('shortlist-candidate');
+        $this->setComponent($component);
+    }
+}
```
Comment (Super-Chama) [inline src/plugins/orangehrmRecruitmentPlugin/Controller/ShortlistCandidateController.php:36]: ?

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+        .update(this.history.id, {note: this.history.note})
+        .then(() => {
+          return this.$toast.updateSuccess();
+        })
+        .then(() => {
+          navigate('/recruitment/viewCandidates');
+        });
+    },
+    onBack() {
+      navigate('/recruitment/vacancy/2/action/1');
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue:246]: can't hardcode ids

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 4 above)
Comment (Ajiharan007) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:109]: no but in the API we only get Id.

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 6 above)
Comment (Ajiharan007) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:173]: text box has capturing two columns

File: src/plugins/orangehrmRecruitmentPlugin/Controller/ShortlistCandidateController.php (hunk 7 above)
Comment (Ajiharan007) [reply src/plugins/orangehrmRecruitmentPlugin/Controller/ShortlistCandidateController.php:36]: we have a shortlist candidate screen

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 2 above)
Comment (Ajiharan007) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:78]: space was already in there

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 6 above)
Comment (Super-Chama) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:173]: use class `--span-column-2` it's globally available

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 5 above)
Comment (Ajiharan007) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:146]: currently we don't implemented it later

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue (hunk 4 above)
Comment (Super-Chama) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: @SandeepaOHRM can't we get the status object? hard-coding in FE doesn't feel right

Comment (SandeepaOHRM) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: Me and Ajiharan discussed this. We do have all these action status in the workflow state machine. getting it from through API feels right for me rather hard-coding like this format.
`status:[
{
'label':'Application Initiated';
'status': 1
},
{
'label':'Shortlisted',
'status':2
},
...
]`

Comment (SandeepaOHRM) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: `status:[
{
'label':'Application Initiated';
'status': 1
},
{
'label':'Shortlisted',
'status':2
},
...
]`

Comment (Super-Chama) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: @Ajiharan007 then please make a mock API and add here.

Comment (SandeepaOHRM) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistCandidateScreen.vue:115]: @Super-Chama yeah for the moment lets go with mock API. I will create a seperate API for this

File: installer/Migration/V5_1_0/translation/bg_BG.yaml [hunk 9]
```diff
@@ -1,21 +1,10 @@
...
-    - { target: 'В списъка', unitId: shortlisted, group: recruitment }
-    - { target: 'Интервю Планирана', unitId: interview_scheduled, group: recruitment }
-    - { target: 'Интервю Приет', unitId: interview_passed, group: recruitment }
-    - { target: 'Интервю Неуспешно', unitId: interview_failed, group: recruitment }
-    - { target: 'Работа предлагам', unitId: job_offered, group: recruitment }
-    - { target: 'Наръчник', unitId: manual, group: recruitment }
-    - { target: 'Online', unitId: online, group: recruitment }
-    - { target: 'Резюме', unitId: resume, group: recruitment }
-    - { target: 'Добави кандидат', unitId: add_candidate, group: recruitment }
+    - { target: 'Кратък списък', unitId: shortlist, group: recruitment }
```
Comment (Super-Chama) [inline installer/Migration/V5_1_0/translation/bg_BG.yaml:2]: Hope these removed keys are not affecting any other screens?

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+      const {candidate, performedAction, vacancy, manager, ...rest} = data;
+      const {firstName, lastName, middleName} = candidate;
+      const fullName = `${manager.firstName} ${manager.middleName} ${manager.lastName}`;
+      this.history = {
+        candidate: `${firstName} ${middleName} ${lastName}`,
+        vacancy: vacancy.title,
+        manager:
+          (manager?.terminationId ? this.$t('general.past_employee') : '') +
+          fullName,
+        performedAction: this.actions.find(({id}) => id === performedAction)
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue:193]: action will come as object with id and label from API recruitment/candidateHistory'. no need to have a array and pick from array.

File: src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+        candidate: '',
+        vacancy: '',
+        manager: '',
+        performedAction: '',
+        performedBy: '',
+        performedDate: '',
+        note: '',
+        statusId: '',
+      },
+      actions: [
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ShortlistHistoryScreen.vue:141]: remove this too. check below comment

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: installer/Migration/V5_1_0/translation/bg_BG.yaml (hunk 9 above)
Comment (Ajiharan007) [reply installer/Migration/V5_1_0/translation/bg_BG.yaml:2]: thanks i just missed it

**Review by Super-Chama — APPROVED** [review]

**Pull request merged**